dts/common/nordic/nrf54h20_eng0.dtsi

```diff
@@ -77,6 +79,12 @@
 			ptr-ext-uicr = <&cpurad_uicr_ext>;
 		};
 
+		ficr: memory@fffe000 {
```

Comment:
Since we use `uicr` for UICRs, wouldn't it be better to use `ficr` here? It's a specific kind of memory. Now I see that also in old SoCs the nodes are named `ficr`, so let's be consistent.

Reply:
I think `memory` fits better (following dt spec and its usage), but using ficr is fine I guess

![image](https://github.com/anangl/zephyr/assets/25011557/78f4d283-0324-4d76-b23c-95d086f9e45e)

Reply:
Technically, *ICRs are just memories, but I think our documentation usually describes them as peripherals.